### PR TITLE
Fix order of configure, allocate and run for conv_op

### PR DESCRIPTION
**Pull request by mdigiorgio**
This patch also provides a fix for depthwise convolution. The OpenCL test was
not passing the group parameter.

### CODE REVIEW

File: caffe2/operators/conv_op_impl.h [hunk 1]
```diff
@@ -20,14 +20,16 @@ bool ConvOp<T, Context>::RunOnDeviceWithOrderNCHW() {
...
   const int M = filter.dim32(0);
-  CAFFE_ENFORCE(
-      C == filter.dim32(1) * group_,
-      "Convolution op: input channels does not match: # of input channels ",
-      C,
-      " is not equal to kernel channels * group:",
-      filter.dim32(1),
-      "*",
-      group_);
+  if (C != group_) {
```
Comment (jerryzh168) [inline caffe2/operators/conv_op_impl.h]: can you revert this change?

File: caffe2/mobile/contrib/opencl/operators/conv_op.cc [hunk 2]
```diff
@@ -90,7 +90,9 @@ bool CLConvOp<T, Activation>::RunOnDevice() {
   LOG(ERROR) << "[C2DEBUG] group: " << group_ << " depthwise: " << depthwise << " grouped: " << grouped;
   CAFFE_ENFORCE(C % group_ == 0);
   CAFFE_ENFORCE(M % group_ == 0);
-  CAFFE_ENFORCE(filter_->dim32(1) == C / group_);
+  if (!depthwise) {
+    CAFFE_ENFORCE(filter_->dim32(1) == C / group_);
```
Comment (jerryzh168) [inline caffe2/mobile/contrib/opencl/operators/conv_op.cc]: This condition should be met regardless whether it is depthwise or not right? Or is this change just for debugging purpose?

**Review by jerryzh168 — APPROVED** [review]
Thanks!

**Comment by jerryzh168** [comment]
please merge after you change the CAFFE_ENFORCE

File: caffe2/mobile/contrib/opencl/operators/conv_op.cc (hunk 2 above)
Comment (mdigiorgio) [reply caffe2/mobile/contrib/opencl/operators/conv_op.cc]: Yes, I'll revert both.
